drivers/ethernet/dsa_ksz8xxx.c

```diff
@@ -89,9 +95,15 @@ static void dsa_ksz8xxx_read_reg(const struct ksz8xxx_data *pdev,
 		.count = 1
 	};
 
-	buf[0] = KSZ8XXX_SPI_CMD_RD | ((reg_addr >> 7) & 0x1F);
+	/*
+  buf[0] = KSZ8XXX_SPI_CMD_RD | ((reg_addr >> 7) & 0x1F);
 	buf[1] = (reg_addr << 1) & 0xFE;
 	buf[2] = 0x0;
+  */
+  buf[0] = KSZ8XXX_SPI_CMD_RD;
+	buf[1] = (reg_addr ) & 0xFF;
```

Comment:
For KSZ8794:
https://ww1.microchip.com/downloads/en/DeviceDoc/KSZ8794CNX-Integrated-4-Port-10-100-Managed-Ethernet-Switch-DS00002134C.pdf
3.5.1.1SPI Client Serial Bus Configuration
Figure 3-7 and 3-8

It looks like STM32 has different LSB/MSB order for the SPI.

Your change will break the already working `ip_k66f` and probably SAM (Atmel)